### PR TITLE
feat(composition): refactor and add kms, execution roles, structured logging

### PR DESCRIPTION
**Pull request by wyvern8**
a bunch of cleanup along with new features for env var security.  todo docs.
https://github.com/nordcloud/serverless-kms-secrets

BREAKING CHANGE: refactor of file structure

### CODE REVIEW

File: generators/app/templates/_package.json [hunk 1]
```diff
@@ -4,12 +4,15 @@
...
     "private": true,
+    "homepage": "<%= homepage %>",
     "devDependencies": {
         "@types/aws-lambda": "^8.10.1",
+        "@types/bunyan": "^1.8.4",
         "@types/faker": "^4.1.1",
         "@types/jest": "^21.1.5",
         "@types/node": "^8.0.47",
         "aws-sdk": "^2.226.1",
+        "bunyan": "^1.8.12",
```
Comment (Neko-Design) [inline generators/app/templates/_package.json]: Need a 'TODO' for the Bunyan Splunk logger, have a working sample finally...

File: generators/package/templates/.envExample-aws [hunk 2]
```diff
@@ -1,3 +1,4 @@
 # function specific env vars override and extend shared .env
 SLS_AWS_STAGE=dev
 SLS_AWS_REGION=ap-southeast-2
+SLS_AWS_KMS_KEY_ID=<%= kmsEncryptionKeyId %>
```
[the remark this answers is not in the document]
Comment (wyvern8) [reply generators/package/templates/.envExample-aws]: good point

File: generators/app/templates/_package.json (hunk 1 above)
Comment (wyvern8) [reply generators/app/templates/_package.json]: i noticed that splunk have a lambda blueprint for capturing cloudwatch logstreams and sending to their HEC.  could be worth centralizing there.  https://www.splunk.com/blog/2017/02/03/how-to-easily-stream-aws-cloudwatch-logs-to-splunk.html

File: generators/package/templates/.envExample-aws (hunk 2 above)
[the remark this answers is not in the document]
Comment (wyvern8) [reply generators/package/templates/.envExample-aws]: aside from this, a little concerned about kms latency.. might test a few approaches.

**Pull request merged**